### PR TITLE
storage node garbage collection design doc

### PR DESCRIPTION
**Pull request by Fadila82**
What: Design doc for garbage collection

Why: To have a garbage collection design doc

Thanks for submitting a PR!

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Could the PR be broken into smaller PRs?
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?

### CODE REVIEW

File: docs/design/garbage-collection.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+Storage nodes will sometimes be temporarily unavailable and will miss delete messages. 
+In these cases, unneeded data is considered garbage.
+
+## Ways to create garbage data:
+- Failed or interrupted upload (we used to delete the previously uploaded segments but since the introduction of order limits, we no longer do this.)
+- Regular upload where the longtail is canceled (e.g. uploading to 90 pieces, but the slowest 10 are cut, so we end up with 80 relevant)
+- Upon deletion
+- Upon replacement
+- When a satellite makes a repair and drops the node
+- When the client stops paying his bills
```
Comment (JessicaGreben) [inline docs/design/garbage-collection.md]: "his" should be "their"

Comment (navillasa) [reply docs/design/garbage-collection.md]: ahh good catch

File: docs/design/garbage-collection.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+1. When a delete object command is issued, the uplink retrieves the list of segments for this object, and Delete is called on each segment. 		
+2. A delete segment request is sent to the satellite. 		
+3. The satellite deletes the segment and sends the order limits for deleting the segment on known online storage nodes.
+4. A delete piece request is sent by the uplink to the corresponding storage node for each addressed order limit.
+
+
+## Design
+### Deletion Process
+- The uplink should send a “DataDeletionReport” to the satellite with the list of pieces and corresponding storage nodes it has been unable to delete
+- The satellite responds with a message indicating if there are still k nodes detaining pieces of the segment. If there are, the delete process has failed.
```
[the remark this answers is not in the document]
Comment (navillasa) [reply docs/design/garbage-collection.md]: I'm not sure about this either

[the remark this answers is not in the document]
Comment (wthorp) [reply docs/design/garbage-collection.md]: The deletion processes was determined to be not necessary for this scope.  I'm resolving this.

**Pull request merged**